GAME_DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,122 @@
+- Find a dim, above ground light source.
+- Search for a cave with exposed coal.
+- Wander into the darkness to find a cave that's bright enough to mine in.
+- Find an above-ground structure that can grant the player coal.
+- Dig straight down until you find coal.
+
+Of course, this problem is completely solved as soon as the player gets coal, as they can now explore caves to find more coal with their newfound torches, but it's important that the player has these options in the first place to prevent them from getting stuck on a seemingly insurmountable barrier. This also gives them more freedom in how they can approach the game. Not one option is better or easier than the other, and if one turns out to be, then it should be allowed to be adjusted to encourage other methods.
+
+### Explain when Needed
+If something progress-related has no obvious explanation, the player will have to rely on a wiki to find out how to progress. One example is the bellows; players won't know to place it next to a furnace and jump on it repeatedly. The issue can be fixed by adding a tooltip that tells them how it works, while leaving out details they can intuitively find out themselves.
+
+Having a mechanic that isn't explained or illuded to makes it an "Invisible Mechanic." These can only be found out through looking at the game's code, wiki reading, or asking a developer, thus giving a disadvantage to casual players. It's best to avoid these at all costs.
+
+### Micro Moments
+These are tiny things the player does in between larger events; examples include:
+- Travelling
+- Parkour
+- Mining
+- Building
+- Crafting
+- Managing inventory
+- Fighting
+
+These moments are extremely important as they let the player mentally rest, so making sure they're as satisfying and consistent as possible is a must.
+
+### Make the World Feel Alive
+To add immersion to the game, creatures should perform behaviors outside of player input, such as hunting, migrating, eating, or sleeping.
+
+# Balancing
+When balancing the game, keep in mind how players might interact with the world, the wildlife, and each other.
+
+### 2OP4ME
+At no point should the player be extremely hard to kill. Armor, tools, accessories, and buffs should be designed around aiding the player in skill-based encounters, not making them win regardless of skill.
+
+### Trade-offs
+If the player is given something to aid them, then it should have an appropriate take-away to balance it.
+
+**Examples:**
+- Using rare resources to create a strong tool.
+- An accessory that heals the player, but takes away energy or some other resource.
+- Enemies have strengths and weaknesses towards particular damage types
+
+### Mob Greifing
+Creatures should not be able to grief players without their input. Destructive creatures are reserved for caves.
+
+# Little Details
+The following contains spoilers for future content. If you are not looking into contributing, you should not read this section.
```

Comment:
I would just put this disclaimer at the top, since I think this partially applies to the entire document, there are examples throughout.

Furthermore I would loosen the wording a bit so it's less strict:
```suggestion
This document is intended for contributors and may contain spoilers.
```